packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -34,6 +35,24 @@ export function FileUploader({
     setFiles,
   } = useFileUploader();
   const [allFileNames, setAllFileNames] = useState<string[]>([]);
+  const [fileStatuses, setFileStatuses] = useState<FileStatuses>([]);
+  const fileStatusesRef = useRef<FileStatuses>([]);
+  // File Previewer global states
+  const [isLoading, setLoading] = useState(false);
+  const [isSuccess, setSuccess] = useState(false);
+  const [percentage, setPercentage] = useState(0);
+
+  useEffect(() => {
+    if (fileStatuses.length === 0) return;
+    const success = fileStatuses.every((status) => status?.percentage === 100);
+    setSuccess(success);
+    setLoading(!success);
+
+    const percentage =
+      fileStatuses.reduce((prev, curr) => prev + (curr?.percentage ?? 0), 0) /
```

Comment:
I added this in, so we can keep track of overall uploads occurring, and the total percentage.